### PR TITLE
Delete fw speedup

### PR DESCRIPTION
**Pull request by jstucke**
Deletion of firmware is really slow if the firmware has thousands of included files (also used during "redo analysis"). This PR tries to speed up the process by optimizing the deletion operations in the database.
- improved SQLAlchemy deletion event listener
- improved delete_firmware method in DB interface
- schema update (used "on cascade delete" in all foreign keys) needed for bulk delete command compatibility
- added alembic script to automate DB migration

In a test with a firmware with ~1.5k files this reduced deletion time by roughly 80%

### CODE REVIEW

File: src/storage/migration/util.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+from alembic import config, script
+from alembic.runtime import migration
+
+from storage.db_connection import AdminConnection
+
+alembic_cfg_path = Path(__file__).parent.parent.parent / 'alembic.ini'
+
+
+def db_needs_migration():
```
Comment (dorpvom) [inline src/storage/migration/util.py:11]: I assume this module is where we need to fix the path-problem regarding starting fact from project root or from `src/` directory.

File: src/storage/migration/versions/221cfef47173_added_oncascade_delte_for_foreign_keys.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision = '221cfef47173'
+down_revision = None
+branch_labels = None
+depends_on = None
+
+
+def upgrade() -> None:
```
Comment (dorpvom) [inline src/storage/migration/versions/221cfef47173_added_oncascade_delte_for_foreign_keys.py:18]: Sure

**Review by dorpvom — CHANGES_REQUESTED** [review]

File: src/storage/migration/util.py (hunk 1 above)
Comment (jstucke) [reply src/storage/migration/util.py]: should work now

File: src/storage/migration/versions/221cfef47173_added_oncascade_delte_for_foreign_keys.py (hunk 2 above)
Comment (jstucke) [reply src/storage/migration/versions/221cfef47173_added_oncascade_delte_for_foreign_keys.py:18]: the script was automatically generated but I can remove those things if you like

**Review by dorpvom — APPROVED** [review]

**Pull request merged**